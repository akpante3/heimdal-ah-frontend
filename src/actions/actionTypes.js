import keyMirror from 'keymirror';

export const ACTIONS = keyMirror({
  OPEN_MODAL: null,
  CLOSE_MODAL: null,
  SET_CURRENT_USER: null,
  SET_CURRENT_USER_FAIL: null,
  LOGOUT_USER: null,
  RESET_PASSWORD_STATE: null,
  RESET_PASSWORD_SUCCESS: null,
  RESET_PASSWORD_FAILED: null,
  RESET_PASSWORD_ERROR: null,
  UPDATE_PASSWORD_SUCCESS: null,
  UPDATE_PASSWORD_FAILED: null,
  UPDATE_PASSWORD_ERROR: null,
});
