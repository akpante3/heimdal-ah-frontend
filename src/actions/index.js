import ResetPassword from './ResetPasswordActions';

export default ResetPassword;
