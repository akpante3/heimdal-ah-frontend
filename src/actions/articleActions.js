// import axios from 'axios';
import { ACTIONS } from './actionTypes';
import errorResponse from '../utils/errorResponse';
import axiosInstance from '../utils/axiosInstance';
import { toggleLoader } from './loaderActions';

export const reportArticle = (reportData = {}, toggleModal, flashMessage) => (dispatch) => {
  const { articleId } = reportData;
  axiosInstance
    .post(`/articles/${articleId}/reports`, reportData)
    .then(() => {
      dispatch(toggleLoader());
      flashMessage();
      setTimeout(() => {
        dispatch(() => {
          toggleModal();
          return {
            type: 'ACTIONS.TOGGLE_MODAL',
          };
        });
      }, 200);
    })
    .catch((errors = {}) => {
      console.log(errors);
      console.log(errors.response);
      const { response = {}, request } = errors;

      const data = {
        dispatch,
        request,
        response,
        errors,
      };
      dispatch(toggleLoader());
      return errorResponse(data);
    });
};
