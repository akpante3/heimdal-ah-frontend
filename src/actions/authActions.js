// import axios from 'axios';
import JWT from 'jsonwebtoken';
import { ACTIONS } from './actionTypes';
import { signupEndpoint } from '../apiEndpoints';
import errorResponse from '../utils/errorResponse';
import configureLocalStorage from '../utils/configureLocalStorage';
import axiosInstance, { setToken } from '../utils/axiosInstance';

export const setAuthUser = (token) => {
  const decodedToken = JWT.decode(token);
  const { userId, username, roleId } = decodedToken;

  configureLocalStorage.setAuthUser(token);
  // remember to set axios auth token here
  setToken(token);

  return {
    type: ACTIONS.SET_AUTH_USER,
    payload: {
      user: { userId, username, roleId },
    },
  };
};

export const signupUser = (signupData = {}, history) => (dispatch) => {
  axiosInstance
    .post(signupEndpoint, signupData)
    .then((response) => {
      const { token } = response.data;
      dispatch(setAuthUser(token));
      return history.push('/');
    })
    .catch((errors = {}) => {
      const { response = {}, request } = errors;

      const data = {
        dispatch,
        request,
        response,
        errors,
      };
      return errorResponse(data);
    });
};
