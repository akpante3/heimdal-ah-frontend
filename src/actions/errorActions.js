import { SET_ERRORS, CLEAR_ERRORS, REMOVE_AN_ERROR } from './actionTypes';

export const setErrors = (payload = {}) => ({
  type: SET_ERRORS,
  payload,
});

export const clearErrors = () => ({
  type: CLEAR_ERRORS,
});

export const removeAnError = errorName => ({
  type: REMOVE_AN_ERROR,
  errorName,
});
