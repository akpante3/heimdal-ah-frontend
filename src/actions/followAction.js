import axios from 'axios';
import { FOLLOW_USER, GET_ALL_FOLLOWERS, UNFOLLOW_USER } from './actionTypes';

// const token = window.localStorage.getItem('access-token');
// const paramsId = window.localStorage.getItem('followerId');
// const followedId = parseInt(paramsId, 10);
const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOjEsInVzZXJuYW1lIjoid2FsZSIsInJvbGVJZCI6MSwiaWF0IjoxNTQ4MzAzNDM5LCJleHAiOjE1NDgzODk4Mzl9.8UP-fhrGZXRUXn-xghL_gh89SBN_bO8bif1J3NM1NB0';

export const getAllFollowers = () => (dispatch) => {
  axios({
    method: 'get',
    url: 'https://heimdal-ah-staging.herokuapp.com/api/v1/users/followers',
    headers: { 'access-token': token },
  })
    .then((followData) => {
      console.log('axios=> get menu', followData);
      dispatch({
        type: GET_ALL_FOLLOWERS,
        payload: followData.data,
      });
    }).catch(err => console.log(err));
};

export const followUser = followedId => (dispatch) => {
  axios({
    method: 'post',
    url: `https://heimdal-ah-staging.herokuapp.com/api/v1/users/${followedId}/follow`,
    headers: { 'access-token': token },
  })
    .then((followData) => {
      console.log('axios=> get menu', followData);
      dispatch({
        type: FOLLOW_USER,
        // payload: followData.data,
        payload: { message: followData.data.message, followedId },
      });
    }).catch(err => console.log(err));
};

export const unfollowUser = followedId => (dispatch) => {
  axios({
    method: 'post',
    url: `https://heimdal-ah-staging.herokuapp.com/api/v1/users/${followedId}/unfollow`,
    headers: { 'access-token': token },
  })
    .then((followData) => {
      console.log('axios=> get menu', followData);
      dispatch({
        type: UNFOLLOW_USER,
        // payload: followData.data,
        payload: { message: followData.data.message, followedId },
      });
    }).catch(err => console.log(err));
};
