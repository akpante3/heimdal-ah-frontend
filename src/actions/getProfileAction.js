import axios from 'axios';
import { GET_ALL_PROFILE } from './actionTypes';

export const getAllProfile = () => (dispatch) => {
  axios
    .get('https://heimdal-ah-staging.herokuapp.com/api/v1/users')
    .then((profileData) => {
      console.log('axios=> get menu', profileData);
      dispatch({
        type: GET_ALL_PROFILE,
        payload: profileData.data,
      });
    }).catch(err => console.log(err));
};
