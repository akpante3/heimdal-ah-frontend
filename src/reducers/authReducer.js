import { SET_AUTH_USER } from '../actions/actionTypes';

const authDefaultState = {
  isAuthenticated: false,
};

const authReducer = (state = authDefaultState, action) => {
  switch (action.type) {
    case SET_AUTH_USER:
      return {
        ...action.payload,
      };

    default:
      return state;
  }
};

export default authReducer;
