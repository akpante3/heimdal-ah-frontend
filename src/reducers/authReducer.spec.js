/* eslint-disable no-undef */
import { ACTIONS } from '../actions/actionTypes';

// Reducer to be tested
import authReducer from './authReducer';

describe('authReducer', () => {
  it('should update the auth sate when a payload is sent and action type is defined', () => {
    const initialState = { isAuthenticated: false, user: {} };
    const action = { type: ACTIONS.SET_AUTH_USER, payload: { userId: 5 } };

    expect(authReducer(initialState, action)).toMatchSnapshot({});
    expect(authReducer(initialState, action)).toEqual({
      isAuthenticated: true,
      user: {
        userId: 5,
      },
    });
  });

  it('should return default state if no action is specified', () => {
    const initialState = { isAuthenticated: false, user: {} };
    const action = { type: '' };
    expect(authReducer(initialState, action)).toMatchSnapshot({});
    expect(authReducer(initialState, action)).toEqual({
      isAuthenticated: false,
      user: {},
    });
  });
});
