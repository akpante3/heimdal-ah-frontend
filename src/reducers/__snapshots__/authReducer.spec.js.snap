// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authReducer should return default state if no action is specified 1`] = `
Object {
  "isAuthenticated": false,
  "user": Object {},
}
`;

exports[`authReducer should update the auth sate when a payload is sent and action type is defined 1`] = `
Object {
  "isAuthenticated": true,
  "user": Object {
    "userId": 5,
  },
}
`;
